src/main/java/com/cybozu/kintone/client/connection/Connection.java

```diff
@@ -184,7 +184,7 @@ public JsonElement request(String method, String apiName, String body) throws Ki
                 is.close();
             }
         } catch (Exception e) {
-            throw new KintoneAPIException("an error occurred while receiving data");
+            throw (KintoneAPIException) e;
```

Comment:
If you are 100% confident `e` is an instance of `KintoneAPIException` or its sub type, why not catch `KintoneAPIException` ?
Otherwise, this cast throw `ClassCastException` at runtime.

Reply:
@exoego I've merge the PR adding `cause` to `KintoneAPIException` constructor from @kmizu into this PR. As I see, his PR solve this problem. Can you tell me your opinion ?

Reply:
@kmizu's change solved the problem definitely 👍

Reply:
@phongnhat19 I think that my PR solved this problem.